scripts/commit_classifier.py

```diff
@@ -381,7 +381,13 @@ def load_user(phid):
 
             if self.git_repo_dir:
                 patch_proc = subprocess.Popen(
-                    ["patch", "-p1", "--no-backup-if-mismatch", "--force"],
+                    [
+                        "git",
+                        "cinnabar",
+                        "fetch",
+                        f"hg::{self.repo_dir}",
+                        diffs[patch.phid],
```

Comment:
Did you run the code? Did it do what you expect? 

`diffs` is a list and `patch.phid` is a string, so this should raise a type error. What we need is hash of the changeset that we just applied to the local Mercurial repository.


BTW. The cod that is applying the patch in git need to be deleted.

Please run the code and ensure it is doing what you expect.

Reply:
Sorry, I was of the understanding that diffs is a dictionary. Is changeset actually diff?

Reply:
Hey @suhaibmujahid 
Sorry, I did not run the code. I was confused as to what arguments I should pass in order to run it.

Reply:
I have read the code in detail and tried to understand it best. I understand how patches are applied to the local mercurial repository. I am still confused as to where can I find the changeset. Is it available in the `patch`? Kindly help me with this @suhaibmujahid.

Reply:
> I am still confused as to where can I find the changeset. 

The changeset is what we applied here:

https://github.com/mozilla/bugbug/blob/f9906057a5281b8913fb5a92edbe73440953581b/scripts/commit_classifier.py#L376-L380

What you need in your command is the changeset ID/hash that we just applied.